File: dotnet/Samples/Translations/Text/TextTranslationService.cs
using LanguageWeaver.Sdk.Common;
using LanguageWeaver.Sdk.Configurations;
using LanguageWeaver.Sdk.Constants;
using LanguageWeaver.Sdk.Translate.Common.Request;
using LanguageWeaver.Sdk.Translate.Common.Result;

namespace LanguageWeaver.Sdk.Samples.Translations.Text;

public class TextTranslationService
{
    public static void Main()
    {
        using var lwClient = new SdkFactory().GetLanguageWeaverClient(new ClientConfiguration());

        TranslateText(lwClient);
        TranslateTextWithLinguisticOptions(lwClient);
    }

    private static void TranslateText(ILanguageWeaverClient lwClient)
    {
        var translateTextRequest = new TranslateTextRequest
        {
            SourceLanguageId = "eng",
            TargetLanguageId = "fra",
            Model = "generic",
            Input = new List<string>
            {
                "The weather is wonderful today!"
            },
            InputFormat = Format.Plain,
            Dictionaries = new List<string>
            {
                // provide list of dictionaries
            }
        };

        TranslateTextResult translateTextResult = lwClient.TranslateText(translateTextRequest);
        Console.WriteLine(translateTextResult.Translation);
    }

    private static void TranslateTextWithLinguisticOptions(ILanguageWeaverClient lwClient)
    {
        Dictionary<string, string> linguisticOptions = new Dictionary<string, string>();
        linguisticOptions.Add("Spelling", "UK");

        var translateTextRequest = new TranslateTextRequest()
        {
            SourceLanguageId = "fra",
            TargetLanguageId = "eng",
            Model = "generic",
            Input = new List<string> { "j'aime cette couleur" },
            LinguisticOptions = linguisticOptions,
            Dictionaries = new List<string>
            {
                // provide list of dictionaries
            },
            InputFormat = Format.Plain
        };

        var translateTextResult = lwClient.TranslateText(translateTextRequest);
        Console.WriteLine(translateTextResult.Translation);
    }
}

File: javascript/backend/samples/translations/text/TextTranslationService.js
const SdkFactory = require("@language-weaver/lw-sdk-js").default;
const {ClientConfiguration, Format, TranslateTextRequest} = require("@language-weaver/lw-sdk-js");

const translateText = async () => {
    try {
        const lwClient = await SdkFactory.getLanguageWeaverClient(new ClientConfiguration());
        const translateTextRequest = new TranslateTextRequest();
        translateTextRequest.sourceLanguageId = "eng";
        translateTextRequest.targetLanguageId = "fra";
        translateTextRequest.model = "generic";
        // provide list of dictionaries
        // translateTextRequest.addDictionary("dictionaryId or name");
        translateTextRequest.addInput("The weather is wonderful today!");
        translateTextRequest.inputFormat = Format.PLAIN;
        const translateTextResult = await lwClient.translateText(translateTextRequest);

        console.log(translateTextResult.translation);
    } catch (e) {
        console.log(e);
    }
}

const translateTextWithLinguisticOptions = async () => {
    try {
        const lwClient = await SdkFactory.getLanguageWeaverClient(new ClientConfiguration());
        const translateTextRequest = new TranslateTextRequest();
        translateTextRequest.sourceLanguageId = "fra";
        translateTextRequest.targetLanguageId = "eng";
        translateTextRequest.model = "generic";
        translateTextRequest.addInput("j'aime cette couleur");
        translateTextRequest.inputFormat = Format.PLAIN;

        const linguisticOptions = new Map();
        linguisticOptions.set("Spelling", "UK");
        translateTextRequest.linguisticOptions = linguisticOptions;

        const translateTextResult = await lwClient.translateText(translateTextRequest);

        console.log(translateTextResult.translation);
    } catch (e) {
        console.log(e);
    }
}

translateText();
// translateTextWithLinguisticOptions();